presto-thrift-connector/src/test/java/com/facebook/presto/connector/thrift/integration/ThriftQueryRunner.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.integration;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.connector.thrift.ThriftPlugin;
+import com.facebook.presto.connector.thrift.location.HostList;
+import com.facebook.presto.connector.thrift.server.ThriftTpchService;
+import com.facebook.presto.spi.HostAddress;
+import com.facebook.presto.testing.QueryRunner;
+import com.facebook.presto.tests.DistributedQueryRunner;
+import com.facebook.swift.codec.ThriftCodecManager;
+import com.facebook.swift.service.ThriftServer;
+import com.facebook.swift.service.ThriftServiceProcessor;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+public final class ThriftQueryRunner
+{
+    public static final int DEFAULT_WORKERS_COUNT = 3;
+    public static final int DEFAULT_THRIFT_NODES_COUNT = 3;
+
+    private ThriftQueryRunner() {}
+
+    public static QueryRunner createThriftQueryRunner(List<ThriftServer> servers)
+            throws Exception
+    {
+        return createThriftQueryRunner(servers, DEFAULT_WORKERS_COUNT);
+    }
+
+    public static QueryRunner createThriftQueryRunner(List<ThriftServer> servers, int workers)
+            throws Exception
+    {
+        List<HostAddress> addresses = servers
```

Comment:
There will just be no hosts to connect to. Technically, it's fine, but an exception will be throw by the connector.